### PR TITLE
Version 1

### PR DESCRIPTION
**Pull request by steven-murray**
This PR implements a bunch of infrastructure changes, including flake8 and black linting, conversion to setuptools_scm for versioning, and slightly better dependency management.

It will take the code to version one (once merged).

### CODE REVIEW

**Comment by steven-murray** [comment]
@MuellerSeb just pinging you that I've requested your review on this :-)

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ignore =
+    E203
+    E266
+    E231
+    E501
+    W503
+    F403
+    F401
+    N803  # I like uppercase single-character names sometimes.
+    N806  # Same here
```
Comment (MuellerSeb) [inline .flake8]: Since it is a rather technical package, I think this is OK.

File: .travis.yml [hunk 2]
```diff
@@ -54,13 +53,20 @@ jobs:
...
-      install: python -m pip install -r requirements_dev.txt
+      install: python -m pip install .[dev] coveralls
       script:
-        - python setup.py install
         - python -m pytest --cov hankel --cov-report term-missing -v tests/
-        # - python -m coveralls
+        - python -m coveralls
       name: Coverage on Linux
 
+    - stage: lint
```
Comment (MuellerSeb) [inline .travis.yml]: I am really curious how this works ;-)

File: docs/conf.py [hunk 3]
```diff
@@ -68,27 +70,12 @@
...
-                              version_file, re.M)
-    if version_match:
-        return version_match.group(1)
-    raise RuntimeError("Unable to find version string.")
 
-# The short X.Y version.
-version = find_version("..","hankel", "__init__.py")
 # The full version, including alpha/beta/rc tags.
-release = find_version("..","hankel", "__init__.py")
+release = get_distribution('hankel').version
```
Comment (MuellerSeb) [inline docs/conf.py]: This seems to be nice and easy. Like it.

File: hankel/__init__.py [hunk 4]
```diff
@@ -1,7 +1,4 @@
 """
-hankel API Summary
```
Comment (MuellerSeb) [inline hankel/__init__.py]: Isn't that needed for the sphinx doc?

File: hankel/hankel.py [hunk 5]
```diff
@@ -273,18 +272,16 @@ def xrange_approx(cls, h, nu, k=1):
...
         """
         r = roots(1, nu)[0]
-        return np.array(
-            [np.pi ** 2 * h * r ** 2 / 2 / k, np.pi * np.pi / h / k]
-        )
+        return np.array([np.pi ** 2 * h * r ** 2 / 2 / k, np.pi * np.pi / h / k])
 
     @classmethod
-    def G(cls, f, h, k=None, *args, **kwargs):
+    def final_term_amplitude(cls, f, h, k=None, *args, **kwargs):
```
Comment (MuellerSeb) [inline hankel/hankel.py]: I like descriptive naming.

File: hankel/hankel.py [hunk 6]
```diff
@@ -298,18 +295,33 @@ def G(cls, f, h, k=None, *args, **kwargs):
...
         -------
-        The value of G.
+        The value of G, the amplitude of the final term in the series' sum.
         """
         if k is None:
             return np.sqrt(2 * h / np.pi) * f(np.pi * np.pi / h)
         return np.sqrt(np.pi / (2 * h)) * f(np.pi * np.pi / h / k)
 
+    @classmethod
+    def G(cls, f, h, k=None, *args, **kwargs):
```
Comment (MuellerSeb) [inline hankel/hankel.py]: Why is it still called `G` here and not `final_term_amplitude`?

File: tests/test_get_h.py [hunk 7]
```diff
@@ -113,7 +116,7 @@ def test_xrange():
     assert x1[1] < x2[1]
 
 
-def test_G():
+def test_g():
```
Comment (MuellerSeb) [inline tests/test_get_h.py]: Wasn't that renamed to `final_term_amplitude`?

**Comment by MuellerSeb** [comment]
Another question is, if we really need a deprecation cycle with version 1.0? Regarding the `G` method.

**Comment by MuellerSeb** [comment]
I am fine with the changes.
When you are adamant that the `G` method should be in v1 with a deprecation round, I am fine with that.
But one could argue, that everything pre-v1 could be seen as beta versions, where this major jump would allow to make changes like that without a deprecation warning.
But you can decide that, I am fine either way ;-)

File: hankel/hankel.py (hunk 6 above)
Comment (steven-murray) [reply hankel/hankel.py]: Good catch. Fixed it up.

File: tests/test_get_h.py (hunk 7 above)
Comment (steven-murray) [reply tests/test_get_h.py]: Yup. Fixed.

**Comment by steven-murray** [comment]
I think you're right that in general, going to a new major version allows breaking API. However, since this is the first version, I want to be a bit extra careful. I've added in proper deprecation warnings. From here on, breaking API should be done only on major versions, and only with 2 minor versions notice. 

I also fixed up some docs that were rendering incorrectly.

**Pull request merged**